packages/neuron-wallet/src/controllers/wallets.ts

```diff
@@ -348,35 +349,91 @@ export default class WalletsController {
   }
 
   @CatchControllerError
-  public static async sendCapacity(params: {
+  public static async sendTx(params: {
+    id: string
+    walletID: string
+    tx: string,
+    password: string
+    description?: string
+  }) {
+    if (!params) {
+      throw new IsRequired('Parameters')
+    }
+    const transaction: TransactionWithoutHash = JSON.parse(params.tx)
+    try {
+      const walletsService = WalletsService.getInstance()
+      const hash = await walletsService.sendTx(
+        params.walletID,
+        transaction,
+        params.password,
+        params.description
+      )
+      return {
+        status: ResponseCode.Success,
+        result: hash,
+      }
+    } catch (err) {
+      logger.error(`sendCapacity:`, err)
+      return {
+        status: err.code || ResponseCode.Fail,
+        message: `Error: "${err.message}"`,
+      }
+    }
+  }
+
+  @CatchControllerError
+  public static async generateTx(params: {
     id: string
     walletID: string
     items: {
       address: string
       capacity: string
     }[]
-    password: string
     fee: string
-    description?: string
+    feeRate: string
   }) {
     if (!params) {
       throw new IsRequired('Parameters')
     }
     try {
       const walletsService = WalletsService.getInstance()
-      const hash = await walletsService.sendCapacity(
+      const tx = await walletsService.generateTx(
         params.walletID,
         params.items,
-        params.password,
         params.fee,
-        params.description
+        params.feeRate,
       )
       return {
         status: ResponseCode.Success,
-        result: hash,
+        result: JSON.stringify(tx),
```

Comment:
Why use `stringify` here?

Reply:
Just a serialization here, can do it or not.

Reply:
Let’s do that only to the api controller later.

Reply:
Removed now.